django/contrib/contenttypes/fields.py

```diff
@@ -263,30 +275,32 @@ class GenericRelation(ForeignObject):
     one_to_many = False
     one_to_one = False
 
-    def __init__(self, to, **kwargs):
-        kwargs['verbose_name'] = kwargs.get('verbose_name', None)
-        kwargs['rel'] = GenericRel(
+    rel_class = GenericRel
+
+    def __init__(self, to, object_id_field='object_id', content_type_field='content_type',
+            for_concrete_model=True, related_query_name=None, limit_choices_to=None, **kwargs):
+
+        kwargs['rel'] = self.rel_class(
             self, to,
-            related_query_name=kwargs.pop('related_query_name', None),
-            limit_choices_to=kwargs.pop('limit_choices_to', None),
+            related_query_name=related_query_name,
+            limit_choices_to=limit_choices_to,
         )
-        # Override content-type/object-id field names on the related class
-        self.object_id_field_name = kwargs.pop("object_id_field", "object_id")
-        self.content_type_field_name = kwargs.pop("content_type_field", "content_type")
-
-        self.for_concrete_model = kwargs.pop("for_concrete_model", True)
 
         kwargs['blank'] = True
         kwargs['editable'] = False
         kwargs['serialize'] = False
+
         # This construct is somewhat of an abuse of ForeignObject. This field
         # represents a relation from pk to object_id field. But, this relation
         # isn't direct, the join is generated reverse along foreign key. So,
         # the from_field is object_id field, to_field is pk because of the
         # reverse join.
         super(GenericRelation, self).__init__(
-            to, to_fields=[],
-            from_fields=[self.object_id_field_name], **kwargs)
+            to, from_fields=[], to_fields=[object_id_field], **kwargs)
```

Comment:
The arguments passed to `from_fields` and `to_fields` have switched? If it's correct, do we need a test for this?